test/gui/test_multiplayer_session_window.py

```diff
@@ -0,0 +1,582 @@
+from randovania.network_common.error import NotAuthorizedForAction
+from randovania.network_common.game_connection_status import GameConnectionStatus
+from randovania.network_common.multiplayer_session import (
+    MultiplayerSessionEntry, MultiplayerUser, User, MultiplayerSessionAction,
+    MultiplayerSessionActions, GameDetails, MultiplayerWorld, MultiplayerSessionAuditLog, MultiplayerSessionAuditEntry,
+    UserWorldDetail,
+)
+from randovania.network_common.session_state import MultiplayerSessionState
+
+
+@pytest.fixture()
+async def window(skip_qtbot) -> MultiplayerSessionWindow:
+    window = MultiplayerSessionWindow(MagicMock(), MagicMock(), MagicMock())
+    skip_qtbot.addWidget(window)
+    window.connect_to_events()
+    return window
+
+
+@pytest.fixture()
+def sample_session(preset_manager):
+    u1 = uuid.UUID('53308c10-c283-4be5-b5d2-1761c81a871b')
+    u2 = uuid.UUID('4bdb294e-9059-4fdf-9822-3f649023249a')
+
+    return MultiplayerSessionEntry(
+        id=1234,
+        name="The Session",
+        worlds=[
+            MultiplayerWorld(
+                name="W1", id=u1, preset_raw=json.dumps(
+                    preset_manager.default_preset_for_game(RandovaniaGame.METROID_PRIME_ECHOES).as_json
+                ),
+            ),
+            MultiplayerWorld(
+                name="W2", id=u2, preset_raw=json.dumps(
+                    preset_manager.default_preset.as_json
+                ),
+            ),
+        ],
+        users_list=[
+            MultiplayerUser(12, "Player A", True, worlds={
+                u1: UserWorldDetail(GameConnectionStatus.InGame,
+                                    datetime.datetime(2019, 1, 3, 2, 50, tzinfo=datetime.timezone.utc))
+            }),
+        ],
+        game_details=None,
+        state=MultiplayerSessionState.SETUP,
+        generation_in_progress=None,
+        allowed_games=[RandovaniaGame.METROID_PRIME_ECHOES],
+    )
+
+
+async def test_on_session_meta_update(preset_manager, skip_qtbot, sample_session):
+    # Setup
+    network_client = MagicMock()
+    network_client.current_user = User(id=12, name="Player A")
+    network_client.server_call = AsyncMock()
+    game_connection = MagicMock(spec=GameConnection)
+    game_connection.executor = AsyncMock()
+    game_connection.pretty_current_status = "Maybe Connected"
+    game_connection.lock_identifier = None
+
+    u1 = uuid.UUID('53308c10-c283-4be5-b5d2-1761c81a871b')
+    u2 = uuid.UUID('4bdb294e-9059-4fdf-9822-3f649023249a')
+
+    initial_session = sample_session
+    second_session = MultiplayerSessionEntry(
+        id=1234,
+        name="The Session",
+        worlds=[
+            MultiplayerWorld(
+                name="W1", id=u1, preset_raw=json.dumps(
+                    preset_manager.default_preset_for_game(RandovaniaGame.METROID_PRIME_ECHOES).as_json
+                ),
+            ),
+        ],
+        users_list=[
+            MultiplayerUser(12, "Player A", True, worlds={u1: "Online"}),
```

Comment:
This should be a `UserWorldDetail` instead of string `Online`

Reply:
... I ran all the tests before pushing, wth